source/configuration_manual/mail_location/imapc/index.rst

```diff
@@ -100,24 +116,39 @@ Connection Settings
 
    **Optimizations**
 
-   ``fetch-bodystructure``
+   ``no-fetch-bodystructure``
+
+     Disable fetching of IMAP ``BODY`` and ``BODYSTRUCTURE`` from the remote
+     server. Instead, the whole message body is fetched to regenerate them.
+
+     .. versionchanged:: v2.4.0;v3.0.0 Earlier versions had a
+                         "fetch-bodystructure" feature, which is now enabled by
+                         default.
+
+   ``no-fetch-headers``
 
-     Allow fetching ``BODY`` and ``BODYSTRUCTURE``
-     ``FETCH BODY.PEEK[HEADER.FIELDS (..)]``.
+     Disable fetching of specific message headers from remote server using the
+     IMAP ``FETCH BODY.PEEK[HEADER.FIELDS(...)]`` command. Instead, the whole
+     header is fetched and the wanted headers are parsed from it.
 
-   ``fetch-headers``
+     .. versionchanged:: v2.4.0;v3.0.0 Earlier versions had a "fetch-headers"
+                         feature, which is now enabled by default.
 
-     Allow fetching specific message headers from remote server.
+   ``no-fetch-size``
 
-     .. versionadded:: 2.2.30
+     Disable fetching of message sizes using the IMAP ``FETCH RFC822.SIZE``
```

Comment:
Looks like there is still one more inconsistency with the above ones. Should be: `sizes from remote server using`.

Oh, and sometimes it says "the remote server" and sometimes without "the". I guess "the" should be there?